api/proto/teleport/externalaudit/v1/externalaudit_service.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.externalaudit.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/externalaudit/v1/externalaudit.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/externalaudit/v1;externalauditv1";
+
+// ExternalAuditService provides methods to manage external audit.
+// It's used only in Teleport Cloud for feature called 'bring your own audit bucket'.
+//
+// There could be multiple external audit resources in cluster, but only single
+// resource can be set as ClusterExternalAudit.
+// The one which is set as ClusterExternalAudit is used by auth.
+//
+// Supporting multiple ExternalAudit as resources allow us to to add resource,
+// and test connection before actually enabling it in cluster.
+service ExternalAuditService {
+  // GetExternalAudit returns the external audit configuration resource.
+  rpc GetExternalAudit(GetExternalAuditRequest) returns (GetExternalAuditResponse);
+
+  // CreateExternalAudit creates a new External Audit resource.
+  rpc CreateExternalAudit(CreateExternalAuditRequest) returns (CreateExternalAuditResponse);
+
+  // DeleteExternalAudit removes External Audit resource.
+  // You cannot remove external audit that is set as 'ClusterExternalAudit'.
+  // First you need to DeleteClusterExternalAudit or use other resource as
+  // ClusterExternalAudit.
+  rpc DeleteExternalAudit(DeleteExternalAuditRequest) returns (google.protobuf.Empty);
+
+  // SetClusterExternalAudit sets external audit to be used in cluster based on
+  // name of external audit.
+  // This method cause Teleport Auth service reload.
+  // Only single resource can be set to active state.
+  rpc SetClusterExternalAudit(SetClusterExternalAuditRequest) returns (SetClusterExternalAuditResponse);
```

Comment:
What do you think of `EnableClusterExternalAudit` and `DisableClusterExternalAudit` rather than `Set`/`Delete`? `Set` might give the impression that it's just some configuration operation, not that it will cause some whole-cluster disruption.